csrc/scheduler/vectorize_helper.cpp

```diff
@@ -1100,84 +984,43 @@ std::vector<std::pair<ProjectedExtent&, IterDomain*>> getContigVectorSizesOf(
       break;
     }
 
-    auto& mapped_extent_PE = mapper.getMappedExtent(root_id);
-    if (mapped_extent_PE.isZero()) {
-      break;
-    }
-
-    vectorizable_dim_sizes.emplace_back(mapped_extent_PE, root_id);
+    product_of_inner_extents = SimplifyingIrBuilder::mulExpr(
+        product_of_inner_extents, getProjectedExtent(root_id));
   }
-  return vectorizable_dim_sizes;
+  return simplifyExpr(product_of_inner_extents);
```

Comment:
Not for this PR, but should we do the following?
1. Use precomputed values to speedup inference, and
2. For all precomputed values, run `simplifyExpr` when they are created. `simplifyExpr` can be invoked in the ctor of precomputed value.